src/webapp/components/sidebar-nav/SidebarNavMenuGroup.tsx

```diff
@@ -1,98 +1,55 @@
-import React, { useEffect } from "react";
+import React from "react";
 import { makeStyles } from "@material-ui/styles";
 import { List, ListItem, Theme, Collapse, Button, colors } from "@material-ui/core";
 import clsx from "clsx";
 import { ExpandLess, ExpandMore } from "@material-ui/icons";
 import { MenuGroup } from "./SidebarNav";
 import SidebarNavMenu from "./SidebarNavMenu";
 import styled from "styled-components";
-import { useLocation } from "react-router-dom";
-
+import FolderIcon from "@material-ui/icons/Folder";
+import { getUrlParam } from "../../utils/helpers";
 interface SidebarNavProps {
     className?: string;
     groupName: string;
-    currentNaVitem: string[];
-    changeCurrentNavItem: (val: string[]) => void;
     menu: MenuGroup;
 }
 
-const SidebarNavMenuGroup: React.FC<SidebarNavProps> = ({
-    menu,
-    groupName,
-    className,
-    currentNaVitem,
-    changeCurrentNavItem,
-}) => {
-    const isCurrentModule = (val: string) => {
-        if (val) {
-            return location.pathname.includes(val);
-        } else {
-            return false;
-        }
-    };
-
-    const isCurrentNavItem = (val: string[]) => {
-        if (isCurrentModule(groupName)) {
-            return true;
-        }
-        if (currentNaVitem && val[0] === groupName) {
-            return true;
-        } else {
-            return false;
-        }
-    };
-
+const SidebarNavMenuGroup: React.FC<SidebarNavProps> = ({ menu, groupName, className }) => {
     const classes = useStyles(menu.level);
-    const location = useLocation();
-    const [openCollapse, setOpenCollapse] = React.useState(isCurrentNavItem(currentNaVitem));
+    const moduleName = getUrlParam("module") || "";
```

Comment:
should the module be fetched from context instead of query param here ?

Reply:
so context setter still isn't working when triggered from sidebar items. 
just as discussed before. you may check on this. branch's latest commit: 
https://github.com/EyeSeeTea/glass-dev/tree/feature/parse-module-orgunit-from-url
pages are loading but context is not updating.